### PR TITLE
feat: add collections to weaviate for curated discovery

### PR DESCRIPTION
**Pull request by mc-jones**
This PR adds `src/14-curated-discovery/06-collections.ts` which are to be used to build out the "curated discovery"  [prototype](https://www.notion.so/artsy/Curated-Discovery-5e43238042014fae853222582a35acd9?pvs=4) 🔒. 

### Considerations
- We likely won't expose the `shortDescription` on the client but It is an interesting, real world, experiment to see how well it summarizes these. 

Screenshot 2024-06-18 at 5 32 15 PM

### CODE REVIEW

File: src/14-curated-discovery/06-collections.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  process.stdout.write("\n")
+}
+
+/**
+ * Return a short description generated by openai using a collection's long description.
+ *
+ * @param description - an artwork collection's long description
+ * @returns a short, one sentence, description
+ */
+async function getShortDescription(description: string) {
```
Comment (mc-jones) [inline src/14-curated-discovery/06-collections.ts:189]: Since we need to utilize the `openai` client I opted to keep this and `batchShortDescriptions` in this file, rather than moving them to `helpers.ts` and passing the client around.

File: src/14-curated-discovery/06-collections.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  collections: GravityArtworkCollection[],
+  batchSize: number = BATCH_SIZE
+) {
+  console.log(`Inserting collection: ${collections.length}`)
+
+  const batches = _.chunk(collections, batchSize)
+  console.log(`Inserting ${batches.length} batches`)
+
+  for (const collectionsBatch of batches) {
+    const shortDescriptions = await batchShortDescriptions(collectionsBatch)
```
Comment (mc-jones) [inline src/14-curated-discovery/06-collections.ts:155]: requires a non-trivial amount of tokens, so run the code with purpose.

File: src/14-curated-discovery/data.md [hunk 3]
```diff
@@ -141,3 +151,36 @@ puts JSON.pretty_generate(partners.map do |p|
...
+jargon, on the assumption that this would perform better with the embedding
+models. We also want to be consistent in the field names we use for genes. For
+example:
+
+- category → **group**
+- genes → **categories**
+
+```ruby
+# Get all marketing collections with a description
+collections = MarketingCollection.where("description is not null and description <> ''").where(published: true)
```
Comment (anandaroop) [inline src/14-curated-discovery/data.md:167]: **suggestion:** We should consider changing this logic…

I was looking at the data dump and noticing all the html markup. This struck me as different from e.g. the artist blurb field that we extract into `artists.json` — there the text is markdown, which implies fewer tokens and is a bit more readable. Honestly I'm not sure how much difference that makes to the embedding model (other than token cost) but I was wondering if we had a similar markdown version of the `description`. 

And the answer is: yes, we also have a `description_markdown`. But the interesting thing (I think I knew this at one point but had to re-learn this just now) is that `description` is a legacy field from when these collections were [migrated](https://artsy.github.io/blog/2022/05/09/deprecating-kaws/) from a dedicated microservice called Kaws. Now that they live in Gravity, it's actually `description_markdown` which is [exposed to admins](https://api.artsy.net/admin/curated_marketing_collection/9e3af5d3-da56-4eaf-89b9-fcf865996e12).

Bottom line, `description_markdown` is the more up-to-date source of truth; covers 90% of collections instead of 80%; and uses fewer tokens to boot.

File: src/14-curated-discovery/types.ts [hunk 4]
```diff
@@ -40,6 +40,17 @@ export type GravityArtwork = {
   partner_name: string
 }
 
+export type GravityArtworkCollection = {
```
Comment (anandaroop) [inline src/14-curated-discovery/types.ts:43]: Same comment as above, I'd vote to call this `GravityMarketingCollection`

**Review by anandaroop — COMMENTED** [review]
Thanks for tackling this — examining the results made me realizing something about the `description` field, so we'll probably want to re-ingest. Details below…

**Review by anandaroop — APPROVED** [review]
Looks great 🎉 

(Just confirming that we still need to upload the revised output to the shared drive, yes?)

**Comment by mc-jones** [comment]
> Looks great 🎉
> 
> (Just confirming that we still need to upload the revised output to the shared drive, yes?)

Yup. Will do right now. 🙏
